### DISCUSSION TITLE
Attempting to temp mount backup, then delete source pool, getting "cannot destroy (source): pool is busy"

### DISCUSSION THREAD

**Discussion by komali2** [openzfs/zfs#14325] [Q&A] [answered]
I was silly and expanded a raid0 pool over time, until it was a monster 14, 14, 18, 18TB beast ready to kill 58TB of data at a single failure. The pool is called `media`

My versions are:

```
zfs-2.1.4-0ubuntu0.1
zfs-kmod-2.1.4-0ubuntu0.1
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy
```

I am clearly naive so I may have gone the wrong route, but my plan was

1. Get two 5TB drives, plug'm in, create new raid0 pool with them, named `backup`
2. Send snapshot of `media` to them (this seems to be fine because only 8TB of space is used so far - trying to correct my mistake early!) 
3. Unmount `media`
4. Mount `backup` in `/mnt/media` (where `media` was mounted)
5. Delete `media` pool, rebuild so it's raidz1 at least (I understand I will be missing 4TB of space each from my larger drives, the intention is this option lets me swap the 14s for 18s down the road, as I understand it)
6. Send updated snapshot (services will be running during this time) back to the new `media` pool
7. Unmount `backup`
8. Mount `media`
9. Be at peace

```
sudo zfs snapshot media@now
sudo zfs send -R media@now | sudo zfs recv -vF backup
```

A day later, I tear down all services pointing at `media`, and do: 

```
sudo zfs snapshot -r media@dec25
sudo zfs send -RI media@now media@dec25 | pv | sudo zfs recv -Fu backup
```

I learn some things about mounting, and then, I *think*, unmount `media`, and mount `backup` in its place. The intention is to have `backup` take over the mountpoint, and services, while I work on destroying `media` and set it up as a raidz1. 

```
sudo zfs unmount -f media
sudo zfs set mountpoint=/media backup
sudo zfs mount backup
```

As of then, my file system was properly mounted. However, I wasn't able to destroy `media`:

```
sudo zpool destroy media
```

```
cannot destroy 'media': pool is busy
```

All answers online check about grepping services to see what's pointing at its mountpoint, but that doesn't make sense for me, as I do want services online pointing at the (old) mountpoint. Just to make sure, though, I set a different mountpoint for `media`:

```
sudo zfs set mountpoint=/test media
sudo zfs get mountpoint media
```

```
NAME   PROPERTY    VALUE       SOURCE
media  mountpoint  /test       local
```

Still, same error. `lsof` doesn't show anything pointing at `/test`, and, anyway, the pool is still unmounted. 

What's happening here? How can I destroy this pool? Do I have smoke in my eyes and actually the original pool is still mounted and serving the directories?

**Comment by komali2** [comment, marked as answer]
restarted and it fixed, lol

**Reply by rincebrain** [reply]
A number of things might have happened, but in general, I would guess something lazily unmounted and the original user wasn't done using it even still, so the pool refused to destroy. Container and virtualization daemons are common culprits for this.

More generally, anything that uses mount namespaces can sometimes hold these open such that it looks like this but is still mounted in the other namespaces, so you could find it by grepping around in /proc/*/mounts.
